src/server/db/datastore/v0_1/types.go

```diff
@@ -0,0 +1,74 @@
+package v0_1
+
+import (
+	"encoding/base64"
+
+	"cloud.google.com/go/datastore"
+	"github.com/google/uuid"
+)
+
+// Version the top-level kind to make parallel loads/queries easier
+const CircuitKind = "v0_1_Circuit"
+const CircuitPermissionKind = "v0_1_CircuitPermission"
+const CircuitContentHeaderKind = "v0_1_CircuitContentHeader"
+const CircuitContentKind = "v0_1_CircuitContent"
+
+// FullCircuit is the top-level model, rarely accessed in its entirety
+type FullCircuit struct {
+	Circuit     Circuit
+	Permissions Permissions
+	Content     []ContentChunk
+}
+
+// Circuit is the top-level fields in a Circuit entity
+type Circuit struct {
+	Name      string
+	Desc      string
+	Version   string
+	Thumbnail string `datastore:",noindex"`
+}
+
+type Permissions struct {
+	LinkID   string
+	LinkPerm int
+	Owner    string
+}
+
+const ContentChunkLimit = 1023 * 1024
+
+// UUID string
+type ContentID string
+
+func NewContentID() ContentID {
+	b, _ := uuid.New().MarshalBinary()
+	return ContentID(base64.URLEncoding.EncodeToString(b))
+}
+
+// Content is a <1023KiB content entry
+type ContentChunk struct {
+	Seq       int
+	ContentID ContentID
+	Content   string `datastore:",noindex"`
+}
+
+func NewContentHeaderKey(circuitKey *datastore.Key) *datastore.Key {
+	return datastore.NameKey(CircuitContentHeaderKind, "Content", circuitKey)
+}
+
+// ContentHeader is the authoratative source for which seqeuence of ContentChunk's is current
+type ContentHeader struct {
```

Comment:
A thought here: If we kept multiple of these objects in the DB, then we can create a coarse history of the circuit.  Not hard to add later on (not a breaking change), but I have to think more if this is sufficient for storing "milestone circuits".

Reply:
Thought about this some more.  We still have only one `ContentHeader` pointing to the "current" version.  We do need 
some kind of timestamp needs to be added to the first `ContentChunk` entry of a given `ContentID` to establish a timeline once `ContentHeader` no longer points to it.